### PR TITLE
Add "share_kubeconfig" config

### PR DESCRIPTION
**Pull request by maciaszczykm**
**What this PR does / why we need it**: Adds possibility if option to share kubeconfig should be available or not. Related to https://github.com/kubermatic/dashboard-v2/pull/857.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: config/kubermatic/values.yaml [hunk 1]
```diff
@@ -58,6 +58,7 @@ kubermatic:
       pullPolicy: "IfNotPresent"
     config: |
       {
+        "share_kubeconfig": false,
```
Comment (p0lyn0mial) [inline config/kubermatic/values.yaml:61]: I know that naming is hard, not sure if this name is better but how about `enable_gen_oidc_kubeconfig`?

**Comment by p0lyn0mial** [comment]
how we are going to enable that feature on dev ? Will you issue a separate pull with changes in `secret` repo ?

File: config/kubermatic/values.yaml (hunk 1 above)
Comment (maciaszczykm) [reply config/kubermatic/values.yaml:61]: It has to be in sync with https://github.com/kubermatic/dashboard-v2/pull/857.

**Comment by maciaszczykm** [comment]
> how we are going to enable that feature on dev ? Will you issue a separate pull with changes in secret repo ?

Yes, like Kristin said in https://github.com/kubermatic/dashboard-v2/pull/857#issuecomment-440668117.

**Review by p0lyn0mial — APPROVED** [review]

**Pull request merged**